app/task_routes.py

```diff
@@ -0,0 +1,108 @@
+from flask.json import jsonify
+from app import db
+from app.models.task import Task
+from flask import Blueprint, request
+from sqlalchemy import desc
+from datetime import datetime
+import os
+from dotenv import load_dotenv
+import requests
+
+load_dotenv()
+
+tasks_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+
+@tasks_bp.route("", methods=["POST"], strict_slashes=False)
+def create_new_task():
+    new_task_data = request.get_json()
+    if new_task_data.keys() >= {"title", "description", "completed_at"}:
+        new_task = Task(**new_task_data)
+        db.session.add(new_task)
+        db.session.commit()
+        return {"task": new_task.task_to_json()}, 201
+    else:
+        return {"details": "Invalid data"}, 400
+
+@tasks_bp.route("", methods=["GET"], strict_slashes=False)
+def get_all_tasks():
+        sort_query = request.args.get("sort")
+        title_query = request.args.get("title")
+        if sort_query == "asc":    
+            tasks = Task.query.order_by(Task.title).all()
+        elif sort_query == "desc":
+            tasks = Task.query.order_by(desc(Task.title)).all()
+        elif sort_query:
+            return {
+                "details": f'Sort by "{sort_query}" is not an option'
+            }, 404
+        elif title_query:
+            tasks = Task.query.filter_by(title=title_query)
+        else:
+            tasks = Task.query.all()
+
+        tasks_response = [task.task_to_json() for task in tasks]
+        return jsonify(tasks_response), 200
+
+def is_int(value):
+    try:
+        return int(value)
+    except ValueError:
+        return False
+
+@tasks_bp.route("/<task_id>", methods=["GET", "PUT", "DELETE"], strict_slashes=False)
```

Comment:
The "" route is split into a function for each method, while this route is not.  For readability, I recommend using a similar pattern throughout your code.  

Just as an aside, combining these methods into one function allowed each method to use the same error checking on lines 54-61 (great code reuse by the way!).  If you wanted to do something similar in a helper function, using the flask abort functionality (https://flask.palletsprojects.com/en/2.0.x/api/) makes it possible to exit from the helper function if either fail state condition is met.